### PR TITLE
Upgrade to RN 0.72.12

### PR DESCRIPTION
**Pull request by zatteo**
## Prerequisites

- [x] Update cozy-client (https://github.com/cozy/cozy-client/pull/1461)
- [x] Update react-native-gzip (https://github.com/cozy/react-native-gzip/pull/3)

## Android

✅ Build
✅ Start

## iOS

✅ Build
✅ Start

## Todo (dev)

- [x] ~~Merge upstream of https://github.com/cozy/react-native-bootsplash to at least 4.X.X (but 5.X.X is already here and stable)~~ (see https://github.com/cozy/cozy-flagship-app/pull/1161/commits/12830f113d678f0d00702ffafdb55017c508c18d)
- [x] Replace `AppState.removeEventListener` call by `.remove()` on the listener returned by `AppState.addEventListener()` https://github.com/cozy/cozy-flagship-app/blob/play-with-rn-upgrade/src/libs/httpserver/HttpServer.ts#L97
- [x] Update react-native-restart to 0.27.0

## Todo (test)

Extensive tests on real devices

**On a real Android**
- [ ] ~~biometry~~
- [x] backup
- [x] share
- [x] send to app
- [x] deep link
- [x] scanner
- [x] enabling geolocation
- [x] CCC

**On a real iPhone**
- [x] biometry
- [x] backup
- [x] share
- [x] send to app
- [x] deep link
- [x] scanner
- [x] enabling geolocation
- [x] CCC

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -46,10 +45,10 @@
       <activity
         android:name=".MainActivity"
         android:label="@string/app_name"
-        android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode"
+        android:configChanges="keyboard|keyboardHidden|orientation|screenLayout|screenSize|smallestScreenSize|uiMode"
```
Comment (Ldoppea) [inline android/app/src/main/AndroidManifest.xml:48]: Maybe we should check how our app behaves on split screens as it seems to be the default configuration now

Cf: https://developer.android.com/guide/topics/large-screens/multi-window-support

**Review by Ldoppea — COMMENTED** [review]
With this upgrade we break support with iOS 11 as the new minimum is 12.4

Before merging this PR we should discuss how we want to handle this (i.e. announcement?)

File: android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (zatteo) [reply android/app/src/main/AndroidManifest.xml:48]: Looks good Tested logging in, navigating between apps and some apps (Drive, Notes, Settings, CoachCO2, mespapiers).

File: android/app/build.gradle [hunk 2]
```diff
@@ -101,37 +71,29 @@ apply from: "./brand.gradle"
...
- * and the benefits of using Hermes will therefore be sharply reduced.
- */
-def enableHermes = project.ext.react.get("enableHermes", false);
-
 android {
     ndkVersion rootProject.ext.ndkVersion
 
     compileSdkVersion rootProject.ext.compileSdkVersion
 
+    namespace "io.cozy.flagship.mobile"
```
[the remark this answers is not in the document]
Comment (zatteo) [reply android/app/build.gradle:96]: I kept this one to be as close as default React Native configuration.

Tested building and running both brand app successfully.

**Pull request merged**